src/PermissionInterface.php

```diff
@@ -119,4 +119,22 @@ public function getRestrictAccess();
    */
   public function setRestrictAccess($access);
 
+  /**
+   * Returns the provider.
+   *
+   * @return string
+   *   The provider.
+   */
+  public function getProvider();
```

Comment:
This documentation is very cryptic. It returns the provider. What is a provider?

Reply:
Ah I get it, this mimics the behaviour of the standard permissions form of Drupal core, where it groups all the permissions by module name.

In the current form this will not be sufficient since this is just a string without any validation. We should probably check that it is a valid module name on save, so that we can retrieve the human readable name from the module handler. In the core permissions form they don't need to check this since they are looping over the `{modulename}.permissions.yml` files which supplies the correct module name for them.

Reply:
It's a bit more difficult in our case because we have 2 kinds of permissions: `GroupContentOperationPermission` and `GroupPermission`. In the case of groups it makes sense to categorize permissions by module, but in the case of group content it would be better to categorize by Entity type or bundle. These permissions are related to the group type, not to the module that happens to define the group type.

This means also that the human readable provider label should be retrieved from the module handler in case of group permissions, and from the EntityTypeBundleInfo service in case of group content permissions.